packages/instantsearch.js/src/lib/utils/__tests__/hydrateRecommendCache-test.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  createSearchClient,
+  createSingleSearchResponse,
+} from '@instantsearch/mocks';
+import algoliaSearchHelper from 'algoliasearch-helper';
+
+import { hydrateRecommendCache } from '../hydrateRecommendCache';
+
+describe('hydrateRecommendCache', () => {
+  it('should hydrate the helper with the recommend cache', () => {
```

Comment:
add a test to ensure no actual recommend request happens if you do a new request with same parameters?